Add a feature to warn if ca_ttl is less than default_svid_ttl * 6

ryysud opened the pull request:
Signed-off-by: Ryuma Yoshida <ryuma.y1117@gmail.com>

<!--
1. If this is your first PR, please read our contributor guidelines
https://github.com/spiffe/spiffe/blob/master/CONTRIBUTING.md
https://github.com/spiffe/spire/blob/master/CONTRIBUTING.md
-->

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
<!-- Please provide a description of the affected functionality -->

This PR affects the functionality of `spire-server run`.

**Description of change**
<!-- Please provide a description of the change -->

I added a feature to warn if ca_ttl is less than default_svid_ttl * 6.
See #1266 for the reason.

**Which issue this PR fixes**
<!-- optional. `fixes #<issue number>` format will close an issue when this PR is merged -->

This PR fixes https://github.com/spiffe/spire/issues/1266.

evan2645 commented on cmd/spire-server/cli/run/run.go:
```diff
@@ -353,6 +354,10 @@ func newServerConfig(c *config) (*server.Config, error) {
 	}
 
+	if !hasExpectedTTLs(sc.CATTL, sc.SVIDTTL) {
+		sc.Log.Warn("ca_ttl is less than default_svid_ttl * 6. SPIRE Server prepares a new CA certificate when 1/2 of the CA lifetime has elapsed in order to give ample time for the new trust bundle to propagate. However, it does not start using it until 5/6th of the CA lifetime. So its normal for an SVID TTL to be capped to 1/6th of the CA TTL. In order to get the expected lifetime on SVID TTLs, the C ...
```
This message is a little long and more detailed than it needs to be I think? I also feel it makes a little more sense to talk about it in terms of the svid_ttl. How about something like this?
`sc.Log.Warn("The configured SVID TTL cannot be guaranteed in all cases - SVIDs with shorter TTLs may be issued if the signing key is expiring soon. Set a CA TTL of at least %v or reduce SVID TTL below %v to avoid issuing SVIDs with a smaller TTL than specified.")`

azdagron commented:
Hi @ryysud! Thanks again for the contribution? Are you planning on continuing work on this PR? If so, I'm happy to keep it open.

ryysud commented:
@evan2645 @azdagron Sorry for the delay...
I addressed PR comments with https://github.com/spiffe/spire/pull/1294/commits/d84533d9967edcf1b1e5b7a9228e06e5d1f692b2 and resolve conflict.  Is my fix correct?

azdagron commented:
This is looking good. I think once the linting failures have been fixed it will be ready to merge.

ryysud commented:
I fixed the linting failures with https://github.com/spiffe/spire/pull/1294/commits/95f987de84f1c32b676b700be4475af737e1b95d.
